docs/development.md

```diff
@@ -48,6 +50,11 @@ cd "$(go env GOPATH)/src/sigs.k8s.io/cluster-api-provider-azure"
 make dep-ensure
 ```
 
+### Get familiar with basic concepts
+
+This provider is modeled after the upstream cluster-api project. To get familiar
+with resources, concepts and conventions refer to the [upstream gitbook](https://kubernetes-sigs.github.io/cluster-api/).
```

Comment:
SUPER DUPER NIT: can we rename it to cluster-api gitbook? Also let's move the link to the bottom of the readme.md for consistency's sake.